src/sources/soundsourceproxy.cpp

```diff
@@ -113,6 +113,19 @@ QList<QDir> getSoundSourcePluginDirectories() {
     return pluginDirs;
 }
 
+QUrl getUrlForTrack(const Track* pTrack) {
+    DEBUG_ASSERT_AND_HANDLE(pTrack != nullptr) {
+        // missing track
+        return QUrl();
+    }
+    const QString canonicalLocation(pTrack->getCanonicalLocation());
+    if (canonicalLocation.isEmpty()) {
+        // file might be missing
```

Comment:
From http://doc.qt.io/qt-5/qurl.html: 
> An empty localFile leads to an empty URL (since Qt 5.4)
What happens in QT 4.8? 

Maybe a comment should clarify this, to protect this code form being optimized out later.